ykllvmwrap/src/jitmodbuilder.cc

```diff
@@ -340,10 +359,10 @@ class JITModBuilder {
         OutlineCallDepth += 1;
         InlinedCalls.push_back(make_tuple(CurInstrIdx, CI));
       }
-      // If this is a recursive call that has been inlined, or if the callee
-      // has the "yk_outline" annotation, remove the inlined code and turn it
-      // into a normal (outlined) call.
-      else if (isRecursiveCall(CF) || CF->hasFnAttribute(YK_OUTLINE_FNATTR)) {
+      // If this is a recursive call or a function attribute says that we cannot
+      // inline the call, then remove the inlined code and turn it into a normal
+      // (outlined) call.
+      else if (isRecursiveCall(CF) || AttrBlocksInlining) {
```

Comment:
Might get a miniscule speed up if we swap these checks around, as `isRecursiveCall` has a small loop inside.

Reply:
Fixed in [27f5741](https://github.com/ykjit/yk/pull/564/commits/27f5741e0fc6ac6b2364b73fa7d51ffe24cddc6a)

This commit should stay separate.